src/win/fs.c

```diff
@@ -1329,12 +1331,32 @@ static void fs__fstat(uv_fs_t* req) {
 
 
 static void fs__rename(uv_fs_t* req) {
-  if (!MoveFileExW(req->file.pathw, req->fs.info.new_pathw, MOVEFILE_REPLACE_EXISTING)) {
-    SET_REQ_WIN32_ERROR(req, GetLastError());
+  int tries;
+  int sys_errno;
+  int result;
+
+  tries = 0;
+
+rename_retry:
+  if (MoveFileExW(req->file.pathw,
+                  req->fs.info.new_pathw,
+                  MOVEFILE_REPLACE_EXISTING) != 0) {
+    SET_REQ_RESULT(req, 0);
     return;
   }
 
-  SET_REQ_RESULT(req, 0);
+  sys_errno = GetLastError();
+  result = uv_translate_sys_error(sys_errno);
+  if ((result == UV_EBUSY || result == UV_EPERM) &&
```

Comment:
The node-graceful-fs code you linked to appears to check for a different error. Is that intentional? 
```
            && (er.code === "EACCES" || er.code === "EPERM")
```

It also appears that it only retries as long as `stat(to)` returns `ENOENT`, fwiw. This was added in https://github.com/isaacs/node-graceful-fs/issues/98 (wherein the delay was also changed from 1 s to 60 s).